File: src/main/java/com/kesherManager/kesherManager/service/TransportService.java
package com.kesherManager.kesherManager.service;

import com.kesherManager.kesherManager.model.Box;
import com.kesherManager.kesherManager.model.Transport;

import java.util.Date;
import java.util.List;
import java.util.Optional;

public interface TransportService {
    // Basic CRUD operations
    List<Transport> getAllTransports();

    Optional<Transport> getTransportById(Long id);

    Transport saveTransport(Transport transport);

    void deleteTransport(Long id);

    // Custom operations
    List<Transport> getTransportsBySourceBox(Box sourceBox);

    List<Transport> getTransportsByDestinationBox(Box destinationBox);

    List<Transport> getTransportsByStatus(Transport.TransportStatus status);

    List<Transport> getTransportsByScheduledDateBetween(Date startDate, Date endDate);

    List<Transport> getTransportsByDestinationType(Transport.DestinationType destinationType);

    List<Transport> getTransportsByCreatedBy(String createdBy);

    List<Transport> getTodayTransports(Transport.TransportStatus status);

    // Update transport status
    Transport updateStatus(Long transportId, Transport.TransportStatus newStatus);

    // Complete a transport
    Transport completeTransport(Long transportId, Date completionDate);
}

File: src/services/transportService.js
import api from './api';

const transportService = {
  // Get all transports
  getAllTransports: async () => {
    const response = await api.get('/api/transports');
    return response.data;
  },

  // Get transport by ID
  getTransportById: async (id) => {
    const response = await api.get(`/api/transports/${id}`);
    return response.data;
  },

  // Create new transport
  createTransport: async (transportData) => {
    const response = await api.post('/api/transports', transportData);
    return response.data;
  },

  // Update existing transport
  updateTransport: async (id, transportData) => {
    const response = await api.put(`/api/transports/${id}`, transportData);
    return response.data;
  },

  // Delete transport
  deleteTransport: async (id) => {
    const response = await api.delete(`/api/transports/${id}`);
    return response.data;
  },

  // Get transports by status
  getTransportsByStatus: async (status) => {
    const response = await api.get(`/api/transports/status/${status}`);
    return response.data;
  },

  // Get transports scheduled for today
  getTodayTransports: async () => {
    const response = await api.get('/api/transports/today');
    return response.data;
  },

  // Complete a transport
  completeTransport: async (transportId, completionData) => {
    const response = await api.patch(`/api/transports/${transportId}/complete`, completionData);
    return response.data;
  },

  // Update transport status
  updateTransportStatus: async (transportId, status) => {
    const response = await api.patch(`/api/transports/${transportId}/status`, { status });
    return response.data;
  }
};

export default transportService;
